src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -506,9 +506,9 @@ CHIP_ERROR MinMdnsResolver::ResolveNodeId(const PeerId & peerId, Inet::IPAddress
         Query query(instanceQName);
 
         query
-            .SetClass(QClass::IN)      //
-            .SetType(QType::ANY)       //
-            .SetAnswerViaUnicast(true) //
+            .SetClass(QClass::IN)       //
+            .SetType(QType::ANY)        //
+            .SetAnswerViaUnicast(false) //
```

Comment:
why do we need this change?

Reply:
TBH, I'm not really sure, but chip-device-ctrl doesn't receive back unicast responses from MDNS. They're transmitted correctly by the device, but they don't show up. Either something else is eating the responses or there's something not quite hooked up correctly, but multicast responses do work. Given that chip-device-ctrl is used for CSG testing, I wanted to make sure this worked for TE3 until we can sort what's up with unicast. The unicast issue is here: https://github.com/project-chip/connectedhomeip/issues/7019

Reply:
Fair enough. Consider adding a TODO to investigate later  - ideally we should try to minimize network traffic and mDNS can be noisy.

Reply:
@cecille would this be also related to #7005